react-ui/src/components/Login/SignIn.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+import Avatar from '@mui/material/Avatar';
+import Button from '@mui/material/Button';
+import CssBaseline from '@mui/material/CssBaseline';
+import TextField from '@mui/material/TextField';
+import Link from '@mui/material/Link';
+import Box from '@mui/material/Box';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import Typography from '@mui/material/Typography';
+import Container from '@mui/material/Container';
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+import { authenticationService } from '../../services/authentication.service';
+import logo from '../../egeria-logo.png';
+
+
+interface Props {
+}
+
+interface State {
+    isLoading: Boolean,
+    errors: Map<string, boolean>,
+    feedbackMessage: String,
+    currentUserValue: any
+}
+
+/**
+ *
+ * React component used for displaying login form.
+ *
+ * @since      0.1.0
+ * @access     public
+ *
+ */
+
+const theme = createTheme();
+
+function Copyright(props: any) {
+    return (
+        <Typography variant="body2" color="text.secondary" align="center" {...props}>
+            {'Powered by : '}
+            <Link color="inherit" href="https://egeria-project.org/">
+                Egeria
+            </Link>
+        </Typography>
+    );
+
+}
+class SignIn extends React.Component<Props, State> {
+
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            isLoading: false,
+            errors: new Map<string, boolean>(),
+            feedbackMessage: 'Sign in',
+            currentUserValue: authenticationService.currentUserValue
+        };
+    }
+
+    handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        const data = new FormData(event.currentTarget);
+        let errors = this.state.errors;
+        let feedbackMessage = this.state.feedbackMessage;
+        let username = new String(data.get('username')).toString();
+        let password = new String(data.get('password')).toString();
+
+        errors.set('username',username ==='');
+        errors.set('password', password ==='');
+
+        this.setState({errors});
```

Comment:
take into account the fact that this is async function, there is a callback after the state has changed

you are changing the state for nothing if on the next line you check the `!Object.values(errors)` right away

Reply:
`this.setState` should always be triggered at the end of all logic, `setState` implies a re-render and a new lifecycle of the component that adapts to its new state.